docs/content/how-to/azure/create-azure-cluster-with-options.md

```diff
@@ -23,6 +23,81 @@ There are a few prerequisites for encrypting the OS disks on the Azure VMs:
 
 After performing these steps, you just need to provide the DiskEncryptionSet ID when creating a hosted cluster.
 
+## Creating Service Principals for Managed Identities
+### Define variables for service principal names
+```
+AZURE_DISK_SP_NAME=<azure-disk-sp-name>
+AZURE_FILE_SP_NAME=<azure-file-sp-name>
+NODEPOOL_MGMT=<nodepool-mgmt-sp-name>
+CLOUD_PROVIDER_SP_NAME=<cloud-provider-sp-name>
+CNCC_NAME=<cncc-sp-name>
+CONTROL_PLANE_SP_NAME=<cpo-sp-name>
+IMAGE_REGISTRY_SP_NAME=<ciro-sp-name>
+INGRESS_SP_NAME=<ingress-sp-name>
+KEY_VAULT_NAME=<kv-name>
+```
+
+### Create service principals and capture app IDs
+```
+DISK_SP_APP_ID=$(az ad sp create-for-rbac --name "${AZURE_DISK_SP_NAME}" --create-cert --cert "${AZURE_DISK_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+FILE_SP_APP_ID=$(az ad sp create-for-rbac --name "${AZURE_FILE_SP_NAME}" --create-cert --cert "${AZURE_FILE_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+NODEPOOL_MGMT_APP_ID=$(az ad sp create-for-rbac --name "${NODEPOOL_MGMT}" --create-cert --cert "${NODEPOOL_MGMT}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+CLOUD_PROVIDER_APP_ID=$(az ad sp create-for-rbac --name "${CLOUD_PROVIDER_SP_NAME}" --create-cert --cert "${CLOUD_PROVIDER_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+CNCC_APP_ID=$(az ad sp create-for-rbac --name "${CNCC_NAME}" --create-cert --cert "${CNCC_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+CONTROL_PLANE_APP_ID=$(az ad sp create-for-rbac --name "${CONTROL_PLANE_SP_NAME}" --create-cert --cert "${CONTROL_PLANE_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+IMAGE_REGISTRY_APP_ID=$(az ad sp create-for-rbac --name "${IMAGE_REGISTRY_SP_NAME}" --create-cert --cert "${IMAGE_REGISTRY_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+INGRESS_APP_ID=$(az ad sp create-for-rbac --name "${INGRESS_SP_NAME}" --create-cert --cert "${INGRESS_SP_NAME}" --keyvault ${KV_NAME} --output json --only-show-errors | jq '.appId' | sed 's/"//g')
+```
+
+### Save service principal IDs and certificate names to a JSON file
+```
+SP_FILE="service-principals.json"
+
+# Save service principal IDs and certificate names to a JSON file
+OUTPUT_FILE="service-principals.json"
+
+cat <<EOF > SP_FILE
+{
```

Comment:
Should this have the control plane vs data plane distinction? Are you planning to reuse this file for workload identity for the managed identities on the data plane side?

Reply:
yep we will use one file for all the managed-identities we will have to pass in